lib/beaker-pe/install/pe_utils.rb

```diff
@@ -101,6 +101,15 @@ def install_ca_cert_on(host, opts)
           end
         end
 
+        # Return just the agent nodes. Removes hosts with roles
+        # master, dashboard, databse, compilemaster, loadbalancer, hub, spoke
+        # @param [Array<Host>] hosts The hosts to install or upgrade PE on
+        #
+        def get_nodes_with_just_agent_role
```

Comment:
@shaigy is there some difference between this method and [`agent_only`](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/roles.rb#L118-L120) that I'm not seeing on first glance that you need here?

Reply:
@kevpl there is difference. agent_only will filter out infrastrcuture nodes where as get_nodes_with_just_agent_role will filter out infrastructure nodes, loadbalancer, hubs and spokes. I am looking for just plane agents that can point to loadbalancer for installation.

Reply:
@shaigy it seems that `agent_only` is more robust than this method. `agent_only` will filter out all of the roles that you've described and more, whereas `get_nodes_with_just_agent_role`'s current logic will only weed out roles that you've hard-coded. It seems that what you want is `agent_only`, not the logic here, since you're saying you want plain agent nodes.

For instance, if the role `mom` is used so a SUT had a roles array of `[agent, mom]`, then this logic would include that SUT whereas `agent_only` would not. This would also be true for any future role names we might make up.